backend/drm/drm.c

```diff
@@ -602,12 +602,26 @@ static bool wlr_drm_connector_set_cursor(struct wlr_output *output,
 		wlr_output_transform_invert(output->transform);
 	wlr_box_transform(&hotspot, transform,
 		plane->surf.width, plane->surf.height, &hotspot);
-	plane->cursor_hotspot_x = hotspot.x;
-	plane->cursor_hotspot_y = hotspot.y;
 
-	if (!update_pixels) {
-		// Only update the cursor hotspot
+	if (plane->cursor_hotspot_x != hotspot.x ||
+			plane->cursor_hotspot_y != hotspot.y) {
+		// Update cursor hotspot
+		conn->cursor_x -= hotspot.x - plane->cursor_hotspot_x;
+		conn->cursor_y -= hotspot.y - plane->cursor_hotspot_y;
+		plane->cursor_hotspot_x = hotspot.x;
+		plane->cursor_hotspot_y = hotspot.y;
+
+		bool ok = drm->iface->crtc_move_cursor(drm, conn->crtc, conn->cursor_x,
+			conn->cursor_y);
+		if (!ok) {
```

Comment:
I don't think this ok variable needs to exist

Reply:
Fixed